### PR TITLE
Tab to select

### PR DESCRIPTION
**Pull request by rickhanlonii**
## Overview

Adds a setting `tabToSelect` to allow tabbing to select suggestion.

Closes https://github.com/moroshko/react-autosuggest/issues/163

## Details

- Adds property `tabToSelect` (defaults to false)
- Sends `Tab` event type to `onSuggestionsUpdateRequested` and `onSuggestionSelected`
- Updates README documentation with setting and event types for above functions
- Adds tests for both true and false values of the setting

### CODE REVIEW

File: src/Autosuggest.js [hunk 1]
```diff
@@ -231,7 +232,24 @@ class Autosuggest extends Component {
             }
             event.preventDefault();
             break;
-
+          case 'Tab': {
+            if (tabToSelect) {
+              event.preventDefault();
```
Comment (rickhanlonii) [inline src/Autosuggest.js]: This may need to be moved inside the if below. Users may want to tab to another field when there's no focused suggestion.

**Comment by moroshko** [comment]
@rickhanlonii Thanks a lot for a detailed PR!

I must admit that I'm a bit confused with the functionality you are trying to achieve. In #163 you mentioned "Slack user mentions". As far as I can see, pressing Tab in Slack is somewhat equivalent to pressing Down, not pressing Enter, i.e. it moves to the next suggestion rather than selecting the focused one. The URL bar in Chrome behaves the same. Am I missing something?

**Comment by rickhanlonii** [comment]
Great point @moroshko, I'll update and re-open.

File: test/plain-list/AutosuggestApp.js [hunk 2]
```diff
@@ -54,6 +54,18 @@ export const onSuggestionsUpdateRequested = sinon.spy(({ value }) => {
   });
 });
 
+export const tabToSelect = {
```
Comment (rickhanlonii) [inline test/plain-list/AutosuggestApp.js]: Note: also going to update this API per the comments [here](https://github.com/moroshko/react-autosuggest/pull/165#discussion_r62430407).

**Pull request closed without merging**